VideoLocker/src/main/java/org/edx/mobile/user/Account.java

```diff
@@ -0,0 +1,201 @@
+package org.edx.mobile.user;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Account {
+    @SerializedName("username")
+    @NonNull
```

Comment:
Gson is performing lots of reflection on runtime anyway in a fairly optimized manner (probably through caching). If we could emulate it’s logic, then we can probably get around any performance bottlenecks for parsing field annotations. Not sure how complicated that would be to implement though. I guess we can leave it for now.
> On 29-Sep-2015, at 12:05 am, Brian Guertin <notifications@github.com> wrote:
> 
> In VideoLocker/src/main/java/org/edx/mobile/user/Account.java <https://github.com/edx/edx-app-android/pull/372#discussion_r40591795>:
> 
> > @@ -0,0 +1,201 @@
> > +package org.edx.mobile.user;
> > +
> > +import android.support.annotation.NonNull;
> > +import android.support.annotation.Nullable;
> > +
> > +import com.google.gson.annotations.SerializedName;
> > +
> > +import java.util.ArrayList;
> > +import java.util.Date;
> > +import java.util.List;
> > +
> > +public class Account {
> > +    @SerializedName("username")
> > +    @NonNull
> It sounded good, but on second thought that's a lot of run-time overhead.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/edx/edx-app-android/pull/372/files#r40591795>.
>